src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1576,63 +1569,101 @@ abstract void onApplicationReply(
 
             private Future<?> signalFuture;
 
-            MqttPublishStream(
+            private int state;
+            private Set<MqttRole> roles;
+
+            MqttServerStream(
                 MessageConsumer application,
                 long routeId,
                 long initialId,
                 long replyId,
                 int packetId,
-                String topicFilter)
+                String topicFilter,
+                Subscription subscription,
+                MqttRole role)
```

Comment:
Suggest moving the `subscription` specific stuff to a method and out of the constructor because the same stream might be used first for publishing.